lasagne/layers/base.py

```diff
@@ -262,30 +234,17 @@ def __init__(self, incomings, name=None):
                 an optional name to attach to this layer
         """
         self.input_shapes = [incoming if isinstance(incoming, tuple)
-                             else incoming.get_output_shape()
+                             else incoming.output_shape
                              for incoming in incomings]
         self.input_layers = [None if isinstance(incoming, tuple)
                              else incoming
                              for incoming in incomings]
         self.name = name
 
-    def get_output_shape(self):
+    @Layer.output_shape.getter
```

Comment:
With `@property` you create a new property replacing the inherited one, with what I used here you overwrite the getter of the inherited property. Both works in this case: as the base class only defines a getter *for now*, we could also just redefine the property as a whole. I guess the question generally boils down to what you want to happen in the subclasses if the base class ever introduces a setter (or deleter): Should they inherit it or not? We'll arguably never introduce a setter for the output shape, so it still doesn't matter which way we choose. I chose this one because it explicitly tells that we're overriding a property that's also present in the base class.